core/deployment/src/main/java/io/quarkus/deployment/dev/RuntimeUpdatesProcessor.java

```diff
@@ -585,6 +587,30 @@ public boolean doScan(boolean userInitiated, boolean forceRestart) {
         }
     }
 
+    /**
+     * This notifies registered extensions of "no-restart" changed files.
+     *
+     * @param noRestartChangedFiles the Set of changed files
+     */
+    public void notifyExtensions(Set<String> noRestartChangedFiles) {
+        if (lastStartIndex == null) {
+            // we don't notify extensions if the application never started
+            return;
+        }
+        scanLock.lock();
+        codeGenLock.lock();
```

Comment:
Why the new locks?

Reply:
For the context, I want to make sure this won't cause problems as you asked for a backport.